src/main/java/spoon/reflect/declaration/CtTypeMember.java

```diff
@@ -30,6 +30,11 @@
 	 * @return declaring class
 	 */
 	@DerivedProperty
-	CtType<?> getDeclaringType();
+	<T> CtType<T> getDeclaringType();
```

Comment:
I needed that to solve another compilation problem in CtConstructorImpl ... but I will try different solution